remove unused utilities

briskt opened the pull request:
### Remove
- Remove runny from Dockerfile and scripts
- Remove s3-expand from Dockerfile

briskt committed:
remove runny

briskt committed:
remove s3-expand

mtompset commented on Dockerfile:
```diff
@@ -15,14 +15,6 @@ RUN mkdir -p /data
 
-# get s3-expand
-RUN curl https://raw.githubusercontent.com/silinternational/s3-expand/1.5/s3-expand -fo /usr/local/bin/s3-expand
-RUN chmod a+x /usr/local/bin/s3-expand
```
Are you sure s3-expand isn't used?

briskt replied:
I investigated. Didn't see any variables that begin with `EXPAND_` defined here or in idp-in-a-box. FWIW, I searched across our GitHub org and found it is used in: mail-admin, developer-portal, and viewipc-terraform (archived).

mtompset commented on application/run-tests-api.sh:
```diff
@@ -3,18 +3,18 @@ set -x
 mkdir -p /data/runtime/mail
 
 # Run database migrations
-whenavail ${MYSQL_HOST} 3306 100 runny /data/yii migrate --interactive=0
```
That was weird having it in the middle here...

mtompset commented on application/run-tests.sh:
```diff
@@ -2,13 +2,13 @@
+composer install --no-interaction --no-scripts
 
 # Try to run database migrations
-runny whenavail db 3306 100 ./yii migrate --interactive=0
```
... and having it at the front here. I do think these runny cleanups are nice.

forevermatt commented on application/run-cron.sh:
```diff
@@ -14,7 +14,7 @@ chown -R www-data:www-data \
 
 # Run database migrations
-runny /data/yii migrate --interactive=0
+/data/yii migrate --interactive=0
```
I forgot that one other feature of `runny` is that it causes the script to exit for non-zero return codes, rather than keep going. We'll probably want something here to cause the script to exit if the migrations fail now.

briskt replied:
I was wondering if I should add the `set -e`. Sounds like that would be important.

forevermatt replied:
Yep. I just wasn't sure what a `set -e` would do here in this run-cron.sh script, with its user of `term_handler()` and such.

briskt replied:
Good question. But FWIW, the `term_handler` is broken now that apache runs in the foreground, which happened when we moved from Logentries to Cloudwatch Logs. I haven't figured out a new solution for that yet.

forevermatt commented on application/run-tests.sh:
```diff
@@ -2,13 +2,13 @@
 
 # Try to run database migrations
-runny whenavail db 3306 100 ./yii migrate --interactive=0
+whenavail db 3306 100 ./yii migrate --interactive=0
```
Similarly here (e.g. exiting if migrations fail)

briskt committed:
add set -e and set -x to test scripts

briskt committed:
run psr2 check in parallel with the tests

briskt committed:
exit run-cron.sh if migration fails

briskt committed:
removed test script line that is no longer needed -- it has been failing

for quite some time, but the addition of the set -e caused it to fail the script

forevermatt commented on application/run-cron.sh:
```diff
@@ -15,6 +15,9 @@ chown -R www-data:www-data \
 # Run database migrations
 /data/yii migrate --interactive=0
+if [ $? -ne 0 ]; then
+    exit $?
```
Can you read the `$?` more than once for a given execution? I simply don't know whether this will work, or whether the second `$?` will be returning something about the `if` comparison.

You may know better/already. Feel free to proceed as you see fit.

briskt replied:
I don't know, actually. That's a great observation.

briskt committed:
ensure the exit uses the desired status --skip-ci
